app/models/concerns/deliverable.rb

```diff
@@ -1,8 +1,8 @@
 module Deliverable
   extend ActiveSupport::Concern
 
-  FINAL_DELIVERY_STATUS = %w[delivered hard_bounce blocked invalid_email error].freeze
   FAILED_DELIVERY_STATUS = %w[soft_bounce hard_bounce blocked invalid_email error].freeze
+  DELIVERED_STATUS = %w[delivered].freeze
```

Comment:
En fait ce sont les seuls statuts qui nous intéressent non ?
Le fait que ce soit `sent`, `accepted` etc. ça ne nous intéresse pas ici non ?
Peut-être qu'alors on devrait prendre que ces statuts en compte dans l'enum `delivery_status`.
Ainsi on filtre les webhooks qui ne concernent pas ces statuts.
Ça résoudrait le problème initial qui amène cette PR non ?

Reply:
Ah mais on avait peur des statuts que l'on louperait c'est pour ça qu'on avait changé ce comportement qui était ce que tu avais fait initialement il me semble

Reply:
Je pense qu'on peut faire un mix des 2. On peut garder ça au niveau niveau des models/concern: 

```ruby 
FAILED_DELIVERY_STATUS = %w[soft_bounce hard_bounce blocked invalid_email error].freeze
DELIVERED_STATUS = %w[delivered].freeze

included do 
  enum delivery_status: (FAILED_DELIVERY_STATUS + DELIVERED_STATUS).index_by(&:itself)
end
```
puisqu'il n'y a que l'info de delivrabilité que l'on veut récupérer ici.

Par contre on peut énumérer la liste de tous les statuts possibles au moment de processer le webhook pour toujours être sûr qu'on ne reçoit pas de statuts inconnu.


WDYT ?

Reply:
Oui tu as raison c'est un bon point. Mais est-ce que dans ce cas-là ce champ doit s'appeler `delivery_status` ? Parce que seuls ces statuts correspondent à la délivrabilité je pense. 
D'ailleurs je me rends compte que je ce que ce que je dis [là](https://github.com/gip-inclusion/rdv-insertion/pull/2100#discussion_r1638423861) est faux avec la mise en place actuelle puisqu'on a une date de délivrance 😢.

Peut-être que ce qu'on peut faire dans de cas c'est: 

* Garder que les statuts de delivery comme ce que je propose précédemment
* Renommer `delivered_at` en `last_brevo_webhook_received_at`
* Toujours mettre à jour `last_brevo_webhook_received_at` pour savoir qu'on a reçu un webhook
* Définir une méthode `def delivered_at = last_brevo_webhook_received_at if delivered?` qui reste envoyée dans le blueprint

Qu'est-ce que tu en penses ? Est-ce que tu penses que les statut `sent`, ou `request` peuvent être utiles à avoir en DB ?
Sinon l'alternative c'est de tout sauvegarder mais du coup parler plutôt de `brevo_status` que de `delivery_status` je pense. J'ai dû mal à voir ce qui est le mieux entre les 2 j'aimerais ton avis

Reply:
> Sinon l'alternative c'est de tout sauvegarder mais du coup parler plutôt de brevo_status que de delivery_status je pense

Pareil ici si on fait on pourrait alors avoir une méthode `def delivery_status = brevo_status if brevo_status.in?(DELIVERY_STATUSES)`

Reply:
> Sinon l'alternative c'est de tout sauvegarder mais du coup parler plutôt de brevo_status que de delivery_status je pense

En fait je ne sais pas si c'est une bonne idée parce qu'il n'y a pas de "statut" brevo j'ai l'impression, brevo n'envoie que des event

Reply:
> Je pense qu'on peut faire un mix des 2. On peut garder ça au niveau niveau des models/concern:
> 
> ```ruby
> FAILED_DELIVERY_STATUS = %w[soft_bounce hard_bounce blocked invalid_email error].freeze
> DELIVERED_STATUS = %w[delivered].freeze
> 
> included do 
>   enum delivery_status: (FAILED_DELIVERY_STATUS + DELIVERED_STATUS).index_by(&:itself)
> end
> ```
> 
> puisqu'il n'y a que l'info de delivrabilité que l'on veut récupérer ici.
> 
> Par contre on peut énumérer la liste de tous les statuts possibles au moment de processer le webhook pour toujours être sûr qu'on ne reçoit pas de statuts inconnu.
> 
> WDYT ?

Plus que le risque de recevoir un status inconnu c'est ce que j'ai mis en commentaire juste avant qui m'embête un peu. Qu'en penses tu ?

Reply:
Effectivement ce sont des event, mettre `brevo_status` porterai à confusion je pense.
J'aime bien l'idée que tu proposes pour le changement sur le `delivered_at`.
Avoir la date de reception du webhook de brevo suffira imo pour un éventuel débug (on ira chercher dans les logs Brevo de toute façon), dans ce cas on a pas besoin de savoir si c'est sent/request... et on peut réduire la liste des status possible aux failed et delivered.

Reply:
done :)
Ca fait des lignes en moins et c'est plus clair. Merci pour ta review !

Reply:
J'en ai profité pour nettoyer un peu les tests et les rendre plus lisibles.